### PR TITLE
hybrid-graphics: wip update to work with new gmux patches

### PR DESCRIPTION
**Pull request by Redecorating**
depends on https://github.com/t2linux/linux-t2-patches/pull/5

### CODE REVIEW

**Comment by AdityaGarg8** [comment]
In the guide having instructions to compile apple-os-set, maybe we can replace that with an easier way to download the pre compiled one from your repo. Something like:

```
sudo mv /boot/efi/EFI/bootx64.efi /boot/efi/EFI/bootx64_original.efi
wget (or curl) the bootloader (probably provide options for silent version as well here)
```

**Comment by AdityaGarg8** [comment]
Also for MBP 16,4, since now we have gmux, can we simply add the .conf file using a Linux ISO, or a third party Linux driver for macOS instead of following the complicated guide?

**Comment by networkException** [comment]
> maybe we can replace that with an easier way to download the pre compiled one from your repo

compiling that yourself should stay a documented option in any case

**Comment by Redecorating** [comment]
> Also for MBP 16,4, since now we have gmux, can we simply add the .conf file using a Linux ISO, or a third party Linux driver for macOS instead of following the complicated guide?

yeah we should be able to simply that section a lot, i'll figure out how best to do that later.

**Comment by Redecorating** [comment]
> In the guide having instructions to compile apple-os-set, maybe we can replace that with an easier way to download the pre compiled one from your repo. Something like:
> 
> 
> 
> ```
> 
> sudo mv /boot/efi/EFI/bootx64.efi /boot/efi/EFI/bootx64_original.efi
> 
> wget (or curl) the bootloader (probably provide options for silent version as well here)
> 
> ```


for apple-set-os i kinda want to see if we can move to:

systemd-boot: use kernel patch for apple-set-os
grub: chainload some.efi, then boot linux (or get a command added into grub itself)
refind: get its apple-set-os command to work

then systemd-boot will just work, grub will just require installation of a package that adds grub config to run that efi file, and refind will just need the user to edit their config

**Comment by AdityaGarg8** [comment]
It would be nice if we could get this merged as well since Linux 6.2 is out

**Comment by Redecorating** [comment]
I don't think we have any kernels released yet that support it.

**Comment by AdityaGarg8** [comment]
Does the main branch of linux-t2-patches support this?

**Comment by AdityaGarg8** [comment]
Cause Ubuntu is goona release a Linux 6.2 soon

File: docs/guides/hybrid-graphics.md [hunk 1]
```diff
@@ -72,83 +69,43 @@ This has been tested on the MacBookPro16,1 and the MacBookPro15,1. The 15,3 and
...
 
 ## MacBookPro16,4
 
-The AMD GPU on MacBookPro16,4 is not yet compatible with Linux. As a workaround :-
+Currently the Radeon 5600M AMD GPU on MacBookPro16,4 is [not working](https://lore.kernel.org/all/3AFB9142-2BD0-46F9-AEA9-C9C5D13E68E6@live.com/) with Linux. As a workaround :-
 
 ### If you are able to edit your kernel command line :-
 
-1. This workaround recommends GRUB as the bootloader. If you want to use some other bootloader, you need to figure out how to get an equivalent of step 4 and 6 working with your bootloader.
+1. Edit the kernel command line of this boot and add the `nomodeset` kernel parameter. This will enable you to access your Linux system in safe graphics.
```
Comment (AdityaGarg8) [inline docs/guides/hybrid-graphics.md:82]: We could use `modprobe apple-gmux force_igd=y` instead of `nomodeset`?

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]

File: docs/guides/hybrid-graphics.md (hunk 1 above)
Comment (Redecorating) [reply docs/guides/hybrid-graphics.md:82]: That won't work if apple-set-os isn't active yet.

Comment (AdityaGarg8) [reply docs/guides/hybrid-graphics.md:82]: Makes sense

**Pull request merged**